### PR TITLE
Add DifferenceMeasure

### PR DESCRIPTION
**Pull request by apdavison**
This is needed initially for an extension to openMINDS/computation covering model validation tests, however it is also a starting point for planned future formalization of data analysis methods.

### CODE REVIEW

File: instances/differenceMeasure/kullbackLeiblerDivergence.jsonld [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    "@id": "https://openminds.ebrains.eu/instances/differenceMeasure/kullbackLeiblerDivergence",
+    "@type": "https://openminds.ebrains.eu/controlledTerms/DifferenceMeasure",
+    "definition": "A measure of how one probability distribution is different from a second, reference probability distribution.",
+    "description": null,
+    "interlexIdentifier": null,
+    "knowledgeSpaceLink": null,
+    "name": "Kullback-Leibler divergence",
+    "preferredOntologyIdentifier": null,
+    "synonym": [
+        "KS-statistic"
```
Comment (lzehl) [inline instances/differenceMeasure/kullbackLeiblerDivergence.jsonld]: KL-statistic ?

File: instances/differenceMeasure/meanSquaredError.jsonld [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    "@context": {
+      "@vocab": "https://openminds.ebrains.eu/vocab/"
+    },
+    "@id": "https://openminds.ebrains.eu/instances/differenceMeasure/meanSquaredError",
+    "@type": "https://openminds.ebrains.eu/controlledTerms/DifferenceMeasure",
+    "definition": "The mean squared difference between two series of values.",
+    "description": null,
+    "interlexIdentifier": null,
+    "knowledgeSpaceLink": null,
+    "name": "z-score",
```
Comment (lzehl) [inline instances/differenceMeasure/meanSquaredError.jsonld]: copy paste error: should be "mean squared error"
does this conflict / duplicate the TypeOfUncertainty term ?

File: instances/terminology/differenceMeasure.jsonld [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+  "@id": "https://openminds.ebrains.eu/instances/terminology/differenceMeasure",
+  "@type": "https://openminds.ebrains.eu/controlledTerms/Terminology",
+  "definition": "A measure of the difference between two things",
+  "description": "This may be a numerical or physical quantity, a set of categories, etc. Examples include 'mean squared error', 't-statistic', 'p-value'.",
+  "interlexIdentifier": null,
+  "knowledgeSpaceLink": null,
+  "name": "measured type",
+  "preferredOntologyIdentifier": null,
+  "synonym": null
+}
```
Comment (lzehl) [inline instances/terminology/differenceMeasure.jsonld]: new line missing at the end of the JSON

**Review by lzehl — CHANGES_REQUESTED** [review]
@apdavison aside the small copy/paste errors I found this seems okay. I just have one question in general: does this conflict / duplicate the TypeOfUncertainty terms (in particular the mean squared error)? should we merge them or can we more clearly differentiate them?

File: instances/differenceMeasure/kullbackLeiblerDivergence.jsonld (hunk 1 above)
Comment (apdavison) [reply instances/differenceMeasure/kullbackLeiblerDivergence.jsonld]: fixed

File: instances/terminology/differenceMeasure.jsonld (hunk 3 above)
Comment (apdavison) [reply instances/terminology/differenceMeasure.jsonld]: fixed

File: instances/differenceMeasure/meanSquaredError.jsonld (hunk 2 above)
Comment (apdavison) [reply instances/differenceMeasure/meanSquaredError.jsonld]: fixed copy-paste error

**Comment by apdavison** [comment]
there's clearly some overlap with TypeOfUncertainty - that seems more narrowly defined than DifferenceMeasure.

**Comment by apdavison** [comment]
We could certainly merge them. What would be the downsides of that, as far as you can see?

**Comment by lzehl** [comment]
@apdavison I don't see a real downside. potentially that some methods are not only meant for uncertainty measurements, but you have a better insight in how far this would be confusing in the QuantitativeValue uncertainties. What do you think?

**Comment by lzehl** [comment]
Merging the TypeOfUncertainty into the DifferenceMeasure may require a KG cleanup / migration in EBRAINS, but that cannot be much and should be doable

**Comment by apdavison** [comment]
Handling this properly would require either supporting a hierarchical structure for controlled terms (with types and sub-types), or providing an enum-type mechanism in the schema template language, where we could specify specific controlled term instances.

What about keeping them separate for now, and then next time a similar situation comes up we decide with Oli and perhaps Tom G. how to handle this in the general case?

**Comment by lzehl** [comment]
@apdavison agreed.

**Comment by apdavison** [comment]
ok to merge?

File: instances/terminology/differenceMeasure.jsonld [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  "@context": {
+    "@vocab": "https://openminds.ebrains.eu/vocab/"
+  },
+  "@id": "https://openminds.ebrains.eu/instances/terminology/differenceMeasure",
+  "@type": "https://openminds.ebrains.eu/controlledTerms/Terminology",
+  "definition": "A measure of the difference between two things",
+  "description": "This may be a numerical or physical quantity, a set of categories, etc. Examples include 'mean squared error', 't-statistic', 'p-value'.",
+  "interlexIdentifier": null,
+  "knowledgeSpaceLink": null,
+  "name": "measured type",
```
Comment (lzehl) [inline instances/terminology/differenceMeasure.jsonld]: I overlooked this one... name is "difference measure"

**Review by lzehl — CHANGES_REQUESTED** [review]

File: instances/terminology/differenceMeasure.jsonld (hunk 4 above)
Comment (apdavison) [reply instances/terminology/differenceMeasure.jsonld]: done

**Review by lzehl — APPROVED** [review]

**Pull request merged**